### PR TITLE
Add portal registration API specs (#1197)

### PR DESCRIPTION
**Pull request by hannes-ucsc**
Swagger spec can be viewed interactively here:

https://editor.swagger.io/?url=https://raw.githubusercontent.com/DataBiosphere/azul/issues/hannes/1197-portal-api-spec/rfc/portal-registration-api.yaml

### CODE REVIEW

**Comment by hannes-ucsc** [comment]
@NoopDog suggests to make Integration polymorphic.

File: rfc/portal-registration-api.yaml [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+        type: string
+        format: uuid
+      portal_name:
+        description: The name of the portal for use in UI elements like links or buttons.
+        type: string
+        maxLength: 50
+      portal_icon:
+        type: string
+        description: A URL to an image to symbolize the portal in a UI
+  Integration:
```
Comment (hannes-ucsc) [inline rfc/portal-registration-api.yaml]: Add fields for portal description, contact email, and organisation name.

File: rfc/portal-registration-api.yaml [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+    # Example: Manifest Integration
+
+    Manifest integration with Azul is for portals that index all data in the HCA and that offer visualization of user-defined subsets of that data. These portals can register with Azul and define a manifest integration for specific types of biological, experimental or analytical entities like samples or data files. As visitors use the Data Browser to interactively refine a selection of these entities, the Data Browser uses this API to query for portals with manifest integrations for the types of entities currently selected and renders UI elements with which the visitor can initiate a transfer to the respective portal. Once a visitor initiates the transfer, the Data Browser first requests Azul to prepare a manifest and when that is complete, trtansfers the visitor to the portal, passing along a URL to the prepared manifest. The manifest contains references to the selected entities along with all or some of the associated metadata, depending on the manifest format specified in the integration. The portal can display analysis results or other visualizations for the entities referenced in the manifest.
```
[the remark this answers is not in the document]
Comment (theathorn) [reply rfc/portal-registration-api.yaml]: Spelling "trtansfers".

**Comment by hannes-ucsc** [comment]
Need to research ability of hard-coding example to work around limitation of generated examples not reflecting `oneOf` correctly.

**Comment by hannes-ucsc** [comment]
I'll hold off on trying to workaround the missing oneOf schemas in the UI examples. There is a ticket currently being worked on to address this: https://github.com/swagger-api/swagger-ui/issues/3803

**Comment by hannes-ucsc** [comment]
Need to add availability negotiation protocol.

Consider adding the ability to limit GetManifestInteraction to a fixed set of entities.

**Comment by hannes-ucsc** [comment]
@danielsotirhos, I left the !fixup commit (https://github.com/DataBiosphere/azul/pull/1225/commits/633a20a4e602a9a7294b488a96b2160a2d4402dd) on this branch but merged this branch into `develop` with that commit squashed out. That commit tells you how you may need to update any copies of the spec you may have made.

**Comment by theathorn** [comment]
@hannes-ucsc What is the official URL to use to refer to the released spec?

**Comment by hannes-ucsc** [comment]
@theathorn I would not consider the spec officially released. Note that the spec still resides in the `rfc` folder of the source tree. I do not feel comfortable releasing the spec until we have fully implemented all endpoints. Until we have done so, the spec should be considered a draft because the implementation may require changes to the spec.

The location of the draft on the develop branch is https://editor.swagger.io/?url=https://raw.githubusercontent.com/DataBiosphere/azul/develop/rfc/portal-registration-api.yaml

**Pull request closed without merging**